build/Dockerfile

```diff
@@ -8,26 +18,28 @@ COPY go.mod go.mod
 COPY go.sum go.sum
 # cache deps before building and copying source so that we don't need to re-download as much
 # and so that source changes don't invalidate our downloaded layer
-RUN go mod download
+# In downstream this needs to be a cached vendor/ folder, so do same upstream
+RUN go mod tidy; go mod vendor
```

Comment:
It does not work. I think it's because go mod tidy will clean up go.mod since there is no references to dependencies in code https://github.com/devfile/devworkspace-operator/pull/380/checks?check_run_id=2308958724

I think the way to make caching work with vendoring:

```
RUN go mod download

# Copy the go source
COPY . .

RUN go mod vendor

# compile
```

but maybe there is a better way

Reply:
it works for me locally, and works in brew too. `go mod download` does not update the vendor folder. Why is `go mod tidy` a bad idea? it guarantees that what you need is actually referenced, and that stale references to versions that no longer exist will be updated to accurate ones, eg., api 2.0.0-old-timestamp is replaced with simpler 2.0.0 which DOES resolve to real content in vendor/ folder.

Reply:
> Why is go mod tidy a bad idea?

because it analyzes source code and remove unused dependencies. Since you do it before copying source code, I have no idea how it works for you locally and in brew.
It does not work for me locally and in Github action :man_shrugging:

Reply:
hmm. must have used my previously cached content to build vendor folder.

So you suggest `go mod download; go mod vendor` is better ?

Reply:
go mod vendor does not work for me without source code.
So, my suggestion is 
```
RUN go mod download # download everything from go.sum
COPY . .
RUN go mod vendor # propagate downloaded dependencies to vendor folder.
```